docs/Permissions/Onchain-Permissioning.md

```diff
@@ -0,0 +1,191 @@
+
+To add or remove nodes: 
+
+1. In the `permissioning-smart-contracts` directory, run the Truffle console: 
+
+    ```bash
+    truffle console 
+    ```
+
+1. Open https://permissioning-tools.pegasys.tech/
+
+1. Enter the [enode URL](../Configuring-Pantheon/Node-Keys.md#enode-url) of the node to be added or removed
+and click the *Process* button. 
+
+1. Click the *Copy to clipboard* button for _Add Enode_ or _Remove Enode_. 
+
+1. Paste the copied command into the Truffle Console. 
+
+   When the transaction is included in a block, the transaction receipt is displayed. 
+
+!!! important 
+    When starting the network, the first node must add itself to the whitelist.  
+
+Added nodes do not attempt to reconnect until peer discovery restarts.  To add a whitelisted node as a peer
+without waiting for peer discovery to restart, use [`admin_addPeer`](../Reference/JSON-RPC-API-Methods.md#admin_addpeer). 
+
+## Display Nodes Whitelist 
+ 
+To display the nodes whitelist, paste the following into the Truffle Console: 
+
+```bash
+Rules.deployed().then(function(instance) {instance.getKeyCount().then(function(txCount) {console.log("size of whitelist: " + txCount); var i=txCount; while(i>=0) {instance.getWhitelistKey(i--).then(function(tx) {console.log(tx)})}});});
+```
+
+## Start Other Network Nodes 
+
+For participating nodes that are not going to add or remove nodes from the whitelist:
+
+1. Start the nodes with permissioning contract options: 
+
+    * [--permissions-nodes-contract-enabled](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-enabled)
+     to enable onhcain permissioning
+    
+    * [--permissions-nodes-contract-address](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-address)
+     set to the address of the Ingress contract in the genesis file (`"0x0000000000000000000000000000000000009999"`)
+
+1. Copy the [enode URL](../Configuring-Pantheon/Node-Keys.md#enode-url) and [have node added to the whitelist](#add-and-remove-nodes-from-the-whitelist).
+
+For participating nodes that are going to add or remove nodes from the whitelist: 
+
+1. Ensure [prerequisites installed](#pre-requisites).
+
+1. Complete [permissioning setup](#onchain-permissioning-setup).
+
+1. Copy the [enode URL](../Configuring-Pantheon/Node-Keys.md#enode-url) and [have node added to the whitelist](#add-and-remove-nodes-from-the-whitelist).
+
+1. Have account for node that will interact with permissioning contracts added as an [admin account](#add-and-remove-admin-accounts). 
+
+## Add and Remove Admin Accounts      
+
+The account that deploys the Rules contract is automatically an Admin account. Only Admin accounts can
+add or remove nodes from the whitelist.
+
+To add Admin accounts, paste the following commands into Truffle Console:
+
+```bash tab="Truffle Console Command"
+Rules.deployed().then(function(instance) {instance.addAdmin("<Admin Account>").then(function(tx) {console.log(tx)});});
```

Comment:
Not sure if this will work if you're running this on a truffle console that did not deploy the Rules contract. You might need to use `Rules.at("<rules-address>").then(function(instance) {instance.addAdmin("<Admin Account>").then(function(tx) {console.log(tx)});});`

Reply:
Fixed